File: ROS/src/simpleRos/simpleRos/service_server.py
import time

import rclpy
from rclpy.node import Node
from std_srvs.srv import SetBool


class Service_server(Node):
    def __init__(self):
        super().__init__("hello_pub")
        self.create_service(SetBool, "setBool", self.setBool_callback)
        self.bool = bool()

    def setBool_callback(self, request : SetBool.Request, response : SetBool.Response):
        self.get_logger().info(f"{request.data}")
        self.get_logger().info(f"{self.bool}")
        if request.data != self.bool:
            self.bool = not self.bool
            response.success = True
            response.message = f"{self.bool} setting sucess"
        else:
            response.success = False
            response.message = f"{self.bool} setting fail"
        time.sleep(5)
        return response

def main():
    rclpy.init()
    node = Service_server()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        node.destroy_node()

if __name__ == "__main__":
    main()

File: ROS/src/simple_ros_cpp/src/service_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_srvs/srv/set_bool.hpp"
#include <chrono>
#include <iostream>

using namespace std;
using namespace std::chrono_literals;

class ServiceServer : public rclcpp::Node
{
public:
    ServiceServer()
        : Node("service_server")
    {
        _server = create_service<std_srvs::srv::SetBool>("setBool", std::bind(&ServiceServer::service_callback, this, std::placeholders::_1, std::placeholders::_2));
    }

private:
    typedef std::shared_ptr<std_srvs::srv::SetBool::Request> BRequest;
    typedef std::shared_ptr<std_srvs::srv::SetBool::Response> BResponse;
    rclcpp::Service<std_srvs::srv::SetBool>::SharedPtr _server;
    bool _bool;
    void service_callback(const BRequest request, BResponse response)
    {
        std::string str = request->data ? std::string("true") : std::string("false");
        RCLCPP_INFO(get_logger(), str.c_str());
        rclcpp::sleep_for(2s);
        if (request->data != _bool)
        {
            _bool = !_bool;
            response->success = true;
            response->message = str + " setting success!!";
        }
        else
        {
            response->success = false;
            response->message = str + " setting fail!!";
        }
    }
};

int main()
{
    rclcpp::init(0, nullptr);
    auto node = std::make_shared<ServiceServer>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
